### PR TITLE
Dont stop Cassandra container before removing it

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/DockerCassandraRule.java [hunk 1]
```diff
@@ -66,9 +66,15 @@ public DockerCassandraRule() {
...
-        cassandraContainer = new GenericContainer<>(
-            new ImageFromDockerfile("cassandra_2_2_10", deleteOnExit)
+        cassandraContainer = new CassandraContainer(tmpFsName);
+    }
+
+    private static class CassandraContainer extends GenericContainer<CassandraContainer> {
+
+        private static final boolean DELETE_ON_EXIT = false;
+
+        public CassandraContainer(String tmpFsName) {
```
Comment (mbaechler) [inline backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/DockerCassandraRule.java]: It makes the reader think (s)he can instantiate it several time but that's not the case.

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/DockerCassandraRule.java [hunk 2]
```diff
@@ -66,9 +66,15 @@ public DockerCassandraRule() {
...
         deleteTmpsFsCmd = client.removeVolumeCmd(tmpFsName);
-        boolean deleteOnExit = false;
-        cassandraContainer = new GenericContainer<>(
-            new ImageFromDockerfile("cassandra_2_2_10", deleteOnExit)
+        cassandraContainer = new CassandraContainer(tmpFsName);
+    }
+
+    private static class CassandraContainer extends GenericContainer<CassandraContainer> {
+
+        private static final boolean DELETE_ON_EXIT = false;
```
Comment (mbaechler) [inline backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/DockerCassandraRule.java]: so now it's further from the usage site, I don't see any benefit

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/DockerCassandraRule.java [hunk 3]
```diff
@@ -105,7 +119,7 @@ public void evaluate() throws Throwable {
                     createTmpsFsCmd.exec();
                     cassandraContainer.apply(base, description).evaluate();
                 } finally {
-                    deleteTmpsFsCmd.exec();
+                    stop();
```
Comment (mbaechler) [inline backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/DockerCassandraRule.java]: how is it related to this PR ?

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
test this please

**Comment by mbaechler** [comment]
no time gain ?

**Pull request closed without merging**